Add key rotation and certificate updates procedures (#921)
italia/eidas-it-wallet-docs: docs/en/x5c-evaluation.rst

- **Federation Exit**: Ensure proper X.509 Certificate revocation during voluntary or supervisory body-initiated federation exit.

Key Rotation Procedures
^^^^^^^^^^^^^^^^^^^^^^^^

Federation entities MUST implement key rotation mechanisms to maintain security posture and respond to security events. This section defines the operational procedures for detecting and responding to superior key rotations and for subordinate key renewal.

Superior Key Rotation
"""""""""""""""""""""""

> Subordinate entities MUST perform periodic verification of their trust chain to detect superior key rotations and ensure continued federation membership.

Reviewer: ```suggestion
Subordinate entities MUST perform periodic verification of their Trust Chain to detect superior key rotations and ensure continued federation membership.
```

**Verification Requirements**

Federation Subordinates MUST:

> 1. Fetch their own Subordinate Statement from immediate superior every 24 hours using the fetch endpoint defined in :ref:`trust-infrastructure:Federation API endpoints`.

Reviewer: ```suggestion
1. Periodically fetch their own Subordinate Statement from immediate superior using the fetch endpoint defined in :ref:`trust-infrastructure:Federation API endpoints`. This SHOULD be made within a time interval not superior that 24 hours.
```

> 2. Verify validity and detect updates related tto the X.509 certificate of the Superior in the ``x5c`` field.

Reviewer 2: ```suggestion
2. Verify validity and detect updates related to the X.509 certificate of the Superior in the ``x5c`` field.
```

- **Federation Exit**: Ensure proper X.509 Certificate revocation during voluntary or supervisory body-initiated federation exit.

Key Rotation Procedures
^^^^^^^^^^^^^^^^^^^^^^^^

> Federation entities MUST implement key rotation mechanisms to maintain security posture and respond to security events. This section defines operational procedures for detecting and responding to superior key rotations and for subordinate key renewal.

Reviewer: ```suggestion
Federation entities MUST implement key rotation mechanisms to maintain security posture and respond to security events. This section defines the operational procedures for detecting and responding to superior key rotations and for subordinate key renewal.
```

Periodic Trust Chain Verification
""""""""""""""""""""""""""""""""""

Subordinate entities MUST perform periodic verification of their trust chain to detect superior key rotations and ensure continued federation membership.

**Verification Requirements**

> Federation subordinates MUST:

Reviewer: ```suggestion
Federation Subordinates MUST:
```

1. Fetch their own Subordinate Statement from immediate superior every 24 hours using the fetch endpoint defined in :ref:`trust-infrastructure:Federation API endpoints`.

> 2. Verify X.509 Certificate validity and detect certificate updates in the ``x5c`` field.

Reviewer: ```suggestion
2. Verify X.509 Certificate validity and detect the X.509 certificate updates in the ``x5c`` field.
```

which X.509? The one about the superior or the one about the subordinate?
The superior cannot update the X.509 of the subordinate without having received an CSR, thereofre I suppose that it is about the superior

Reviewer 3: A superior entity cannot trigger a Subordinate Key Rotation. Subordinate Key Rotation mechanism is always started by the Subordinate entity. We are going to add specific clarifications about that.

3. Check for revocation or suspension status.
4. Update local Entity Configuration when superior certificates change.

**Detection Mechanisms**

Subordinates MAY use one or more of the following mechanisms to detect changes:

- **CRL/OCSP Verification**: Monitor Certificate Revocation Lists or Online Certificate Status Protocol responses to detect revocation of superior certificates as defined in :ref:`trust-infrastructure:X.509 Certificate Revocation`. Certificate revocation indicates superior key rotation is in progress.
- **Fetch Endpoint**: Retrieve Subordinate Statement via ``GET {superior}/fetch?sub={subordinate_entity_id}`` to obtain current certificates.
- **Events Endpoint**: Monitor ``GET {superior}/federation_subordinate_events_endpoint?sub={subordinate_entity_id}`` for ``jwks_update`` events. The Federation Subordinate Events endpoint returns historical track of registration events as defined in :ref:`trust-infrastructure:Federation API endpoints`.

**Implementation**

> 

Reviewer: ```suggestion
```
